interpreter/src/lib.rs

```diff
@@ -266,4 +266,44 @@ mod tests {
             assert_eq!(res, error.into(), "{}", name);
         }
     }
+
+    #[test]
+    fn test_type_fn_basic() {
+        let ctx = Context::default();
+
+        // type(1) == 'int'
+        let prog = Program::compile("type(1) == 'int'").unwrap();
+        let result = prog.execute(&ctx).unwrap();
+        assert_eq!(result, Value::Bool(true));
+
+        // type('a') == string
+        let prog = Program::compile("type('a')").unwrap();
+        let result = prog.execute(&ctx).unwrap();
+        assert_eq!(result, Value::String("string".to_string().into()));
+
+        // type(true) == bool
+        let prog = Program::compile("type(true)").unwrap();
+        let result = prog.execute(&ctx).unwrap();
+        assert_eq!(result, Value::String("bool".to_string().into()));
+
+        // type(1) == type('a')
+        let prog = Program::compile("type(1) == type('a')").unwrap();
+        let result = prog.execute(&ctx).unwrap();
+        assert_eq!(result, Value::Bool(false));
+
+        // type(type(1)) == 'string'
+        let prog = Program::compile("type(type(1)) == 'string'").unwrap();
+        let result = prog.execute(&ctx).unwrap();
+        assert_eq!(result, Value::Bool(true));
+
+        // type(type(1)) == type(type(2))
+        let prog = Program::compile("type(type(1)) == type(type(2))").unwrap();
+        let result = prog.execute(&ctx).unwrap();
+        assert_eq!(result, Value::Bool(true));
+
+        // type(type(1)) == type(type("a"))
+        let prog = Program::compile("type(type(1)) == type(type('a'))").unwrap();
+        let result = prog.execute(&ctx).unwrap();
+        assert_eq!(result, Value::Bool(true));
```

Comment:
Can all of these test cases be added to the tests in the `functions.rs` file in the `test_type` tests?